vector/src/main/java/im/vector/app/AutoRageShaker.kt

```diff
@@ -261,7 +259,14 @@ class AutoRageShaker @Inject constructor(
             return
         }
         this.currentActiveSessionId = sessionId
-        this.detector.enabled = _enabled
+
+        hasSynced = session.hasAlreadySynced()
+        session.getSyncStatusLive()
+                .asFlow()
+                .onEach {
+                    hasSynced = it !is SyncStatusService.Status.Progressing
```

Comment:
This is not exaclty what we want no? Looks a bit strange. Also we are subscribing to the flow for the all session lifecycle no?

Reply:
"For the lifetime of the session, confirm we are not in an initial sync", Being outside an initial sync I thought would be reasonably equivalent to after an initial sync.

Web uses the sync token, so equivalent to `hasAlreadySynced`. We could keep a reference to the active sessions and call `hasAlreadySynced` synchronously in `enabled`?

Reply:
@ganfra or @bmarty what would be the best way to check that an initial sync has already been done?

Reply:
I think this is acceptable.
Maybe we should rename `SyncStatusService.Status.Progressing` to `SyncStatusService.Status.InitialSyncProgressing` for clarity.

It's historical, previously we did not track the incremental sync status.